dev-db/etcd-wrapped/files/etcd-wrapped.ebuild

```diff
@@ -0,0 +1,97 @@
+#
+# Copyright (c) 2016 CoreOS, Inc.. All rights reserved.
+# Distributed under the terms of the GNU General Public License v2
+# $Header:$
+#
+
+EAPI=6
+
+inherit systemd
+
+DESCRIPTION="etcd Wrapper"
+HOMEPAGE="https://github.com/coreos/etcd"
+KEYWORDS="amd64 arm64"
+
+LICENSE="Apache-2.0"
+IUSE="etcdctl"
+
+VERSION=${PV}
+VERSION=${VERSION/_alpha/-alpha.}
+VERSION=${VERSION/_beta/-beta.}
+
+case ${VERSION} in
+0*)
+	SLOT="0"
+	;;
+2*)
+	SLOT="2"
+	;;
+3*)
+	SLOT="3"
+	;;
+*)
+	die "Unknown etcd version ${VERSION}"
+	;;
+esac
+
+
+SLOT_MANGLED=${SLOT/0/}
+IMG_FILENAME="etcd-v${VERSION}-linux-${ARCH}.aci"
```

Comment:
Since IMG_FILENAME helps populate SRC_URI don't encode arch using the $ARCH variable as that value is set at runtime. SRC_URI will need to specify all architectures and just guard the correct ones with "amd64? ( ... ) arm64? (...)" syntax